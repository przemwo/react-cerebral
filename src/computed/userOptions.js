import { compute } from 'cerebral';
import { state } from 'cerebral/tags';

import { NOVALUE } from 'common/components/Select';

const DEFAULT_OPTION = {
  value: NOVALUE,
  label: ''
};

function userOptions (usersKeys, get) {
  return (usersKeys || []).reduce(
    (options, userKey) => {
      options.push({
        value: userKey,
        label: get(state`repos.usersById.${userKey}.login`)
      });

      return options;
    },
    [DEFAULT_OPTION]
  );
}

export default compute(
  state`repos.usersById.*`,
  userOptions
);
