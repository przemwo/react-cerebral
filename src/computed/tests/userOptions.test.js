import { runCompute } from 'cerebral/test';

import userOptions from '../userOptions';

describe('userOptions() computed', () => {
  it('returns correct options for users select', () => {
    const fakeState = {
      repos: {
        usersById: {
          4643: {
            id: 4643,
            login: 'garth'
          },
          36270: {
            id: 36270,
            login: 'Guria'
          }
        }
      }
    };
    const result = runCompute(userOptions, { state: fakeState });

    expect(result).toMatchSnapshot();
  });
});
