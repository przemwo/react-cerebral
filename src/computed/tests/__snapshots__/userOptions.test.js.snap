exports[`userOptions() computed returns correct options for users select 1`] = `
Array [
  Object {
    "label": "",
    "value": "__NOVALUE__",
  },
  Object {
    "label": "garth",
    "value": "4643",
  },
  Object {
    "label": "Guria",
    "value": "36270",
  },
]
`;
